http-client-javanet/src/main/java/io/micronaut/http/client/javanet/DefaultJavanetHttpClientRegistry.java

```diff
@@ -0,0 +1,325 @@
+                configuration = this.beanContext.findBean(
+                    HttpClientConfiguration.class,
+                    Qualifiers.byName(clientId)
+                ).orElse(defaultHttpClientConfiguration);
+            } else {
+                configuration = defaultHttpClientConfiguration;
+            }
+
+            if (clientId != null) {
+                loadBalancer = loadBalancerResolver.resolve(clientId)
+                    .orElseThrow(() ->
+                        new HttpClientException("Invalid service reference [" + clientId + "] specified to @Client"));
+            }
+
+            String contextPath = null;
+            if (StringUtils.isNotEmpty(path)) {
+                contextPath = path;
+            } else if (StringUtils.isNotEmpty(clientId) && clientId.startsWith("/")) {
+                contextPath = clientId;
+            } else {
+                if (loadBalancer != null) {
+                    contextPath = loadBalancer.getContextPath().orElse(null);
+                }
+            }
+
+            JavanetHttpClient client = buildClient(loadBalancer, clientKey.httpVersion, configuration, clientId, contextPath, beanContext);
+
+            final JsonFeatures jsonFeatures = clientKey.jsonFeatures;
+            if (jsonFeatures != null) {
+                List<MediaTypeCodec> codecs = new ArrayList<>(2);
+                MediaTypeCodecRegistry codecRegistry = client.getMediaTypeCodecRegistry();
+                for (MediaTypeCodec codec : codecRegistry.getCodecs()) {
+                    if (codec instanceof MapperMediaTypeCodec mapper) {
+                        codecs.add(mapper.cloneWithFeatures(jsonFeatures));
+                    } else {
+                        codecs.add(codec);
+                    }
+                }
+                if (codecRegistry.findCodec(MediaType.APPLICATION_JSON_TYPE).isEmpty()) {
+                    codecs.add(createNewJsonCodec(this.beanContext, jsonFeatures));
+                }
+                client.setMediaTypeCodecRegistry(MediaTypeCodecRegistry.of(codecs));
+            }
+            return client;
+        });
+    }
+
+    private JavanetHttpClient buildClient(
+        LoadBalancer loadBalancer,
+        HttpVersionSelection httpVersion,
+        HttpClientConfiguration configuration,
+        String clientId,
+        String contextPath,
+        BeanContext beanContext
+    ) {
+        ConversionService conversionService = beanContext.getBean(ConversionService.class);
+        return new JavanetHttpClient(
+            loadBalancer,
+            httpVersion,
+            configuration,
+            contextPath,
+            mediaTypeCodecRegistry,
+            beanContext.findBean(RequestBinderRegistry.class).orElseGet(() ->
+                new DefaultRequestBinderRegistry(conversionService)
+            ),
+            clientId,
+            conversionService,
+            beanContext.findBean(JavanetClientSslBuilder.class).orElseGet(() ->
```

Comment:
Replace these manual bean lookups with the `BeanProvider<JavanetClientSslBuilder>` injected into the constructor

Reply:
Done in https://github.com/micronaut-projects/micronaut-core/pull/8441/commits/e7a4fa8b2124bdc2af9aa0dd2ac6d78284f15e0b

There's no lazy orElse for BeanProvider, which means we will create a default each time, even if we don't need it